src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -3512,26 +3512,16 @@ public static ref readonly TensorSpan<T> StackAlongDimension<T>(scoped ReadOnlyS
         /// <param name="x">The <see cref="TensorSpan{T}"/> to take the standard deviation of.</param>
         /// <returns><typeparamref name="T"/> representing the standard deviation.</returns>
         public static T StdDev<T>(in ReadOnlyTensorSpan<T> x)
-            where T : IFloatingPoint<T>, IPowerFunctions<T>, IAdditionOperators<T, T, T>, IAdditiveIdentity<T, T>
+            where T : IFloatingPoint<T>, IAdditionOperators<T, T, T>, IAdditiveIdentity<T, T>, IRootFunctions<T>
         {
             T mean = Average(x);
             Span<T> span = MemoryMarshal.CreateSpan(ref x._reference, (int)x._shape._memoryLength);
             Span<T> output = new T[x.FlattenedLength];
             TensorPrimitives.Subtract(span, mean, output);
             TensorPrimitives.Abs(output, output);
-            TensorPrimitives.Pow((ReadOnlySpan<T>)output, T.CreateChecked(2), output);
-            T sum = TensorPrimitives.Sum((ReadOnlySpan<T>)output);
+            T sum = TensorPrimitives.SumOfSquares((ReadOnlySpan<T>)output);
```

Comment:
This is not correct. Casting `ReadOnlyTensorSpan` to `ReadOnlySpan` can potentially drop elements. I think you will have to implement it by yourself.

Reply:
output is a `Span<T>`. I guess it is the `Subtract` operations that "transforms" the values from a ROTensor into the Span accordingly.

Reply:
The input `ROTensorSpan` can be longer than `ROSpan`. If you cast it directly, any element after the index `2^31-1` will be dropped.

Reply:
1. `x` is ROTensorSpan.
2. `span` is `ROSpan` constructed from `x` (code not changed in this PR).
3. `output` is `Span`, this is used as intermediare during calculations.

Is it step 2 you mean that bug lies? Perhaps for non-contiguous and sparse spans? I'm not 100% familiar with how Tensors work

Reply:
Okay. Didn't notice it's already casting to `Span`. Then I think it's fine.

Reply:
Yes technically this is a bug, albeit a pre-existing one. You can either fix it here, or fix it independently.

Creating a span directly over a `ROTensorSpan` is dangerous due to the potential for there to be more than `2^31` contiguous elements and due to the potential that it represents a "slice" of memory and therefore sparse data.

I imagine this fails, for example, if you create a `4x4` matrix and slice out the upper `2x2` corner. Equally it should fail for a `1xN` input where the stride between elements is `2`.

This should rather be using `Tensor.Subtract(x, mean, output)` (where `output` is created with the correct shape). Then `Tensor.Abs(output, output)`, and then there should likely be a proper `Tensor.SumOfSquares` API that does the right thing itself.

The implementation of any `TensorSpan` handler then basically needs to do the "right" thing which typically just involves determining if the buffer is contiguous, dense, and less than 2^31 elements. If all of these are met, it can simply create a single span over the raw data and pass that into `TensorPrimitives`; otherwise, it needs to slice the largest contiguous buffer it can and effectively operate "per row" (or per 2^31 elements).

Reply:
This isn't necessary to fix here, but it does need an issue logged and needs to be fixed for .NET 10

CC. @michaelgsharp as this and other similar issues probably exist in the experimental API set that got shipped.